test/testdata/rbi/array.rb

```diff
@@ -70,3 +70,6 @@
 
 T.reveal_type(arr.fetch(0, 'error')) # error: Revealed type: `T.any(Integer, String)`
 T.reveal_type(arr.fetch(0) { 'error' }) # error: Revealed type: `T.any(Integer, String)`
+
+x = [[1, "a"], [2, "b"]]
+T.reveal_type(x.transpose) # error: Revealed type: `[T::Array[Integer], T::Array[String]] (2-tuple)`
```

Comment:
Can you add a test case for when the arrays are not the same length? I think we need to make some sneaky `nilable` things show up here.